pkg/tms/tms.go

```diff
@@ -118,6 +118,7 @@ func NewCommunicationInstance(httpClient piperHttp.Uploader, tmsUrl, uaaUrl, cli
 		return communicationInstance, errors.Wrap(err, "Error fetching OAuth token")
 	}
 	log.RegisterSecret(token)
+	log.RegisterSecret(clientSecret)
```

Comment:
I would do it here instead:
https://github.com/SAP/jenkins-library/blob/master/cmd/tmsUpload.go#L87

And here:
https://github.com/SAP/jenkins-library/blob/master/pkg/tms/tms.go#L133

Reply:
yes, should be called as early as possible. Done

Reply:
OK, added

Reply:
Thank you!